packet/conn.go

```diff
@@ -331,41 +307,45 @@ func (c *Conn) WritePacket(data []byte) error {
 }
 
 func (c *Conn) writeCompressed(data []byte) (n int, err error) {
-	var compressedLength, uncompressedLength int
-	var payload, compressedPacket bytes.Buffer
-	var w io.WriteCloser
-	minCompressLength := 50
-	compressedHeader := make([]byte, 7)
+	var (
+		compressedLength, uncompressedLength int
+		payload                              *bytes.Buffer
+		compressedHeader                     = make([]byte, 7)
+	)
 
-	switch c.Compression {
```

Comment:
moving this switch so that it only creates a compression writer if `len(data) > MinCompressionLength` otherwise all the allocations that come with creating the writer happen only to have the writer thrown away because compression won't occur.